[FIX] [15.0] account_invoice_supplier_self_invoice: New partner billing prefix

Shide opened the pull request:
Use a new partner self billing prefix to not use the `partner.ref` field.
Added a button in the partner to create both sequences before the first self bill invoice (maybe you want to modify the sequence)

Please review if you want @etobella @rafaelbn 

MT-906 @moduon

Shide committed:
[FIX] account_invoice_supplier_self_invoice: Use new partner billing code

rafaelbn reviewed (approved):
Tested in runboat 👍

rafaelbn commented:
Could I recommend, please:

- [x] Make visible the button for sending the invoice in draft to the vendor

Why? Because for our general case, we send it in draft, the vendor confirm it by replying and then we validate.

A [requirement](https://www.sumup.com/es-es/facturas/glosario/facturacion-por-el-destinatario/) is

> Por cada factura emitida, el empresario o profesional ha de mostrar su aceptación, mediante un procedimiento que acordará previamente con su cliente. 

Thanks!

etobella commented on account_invoice_supplier_self_invoice/views/res_partner_views.xml:
```diff
@@ -7,16 +7,28 @@
                     <field
                         name="self_invoice_sequence_id"
                         readonly="1"
-                        attrs="{'invisible': [('self_invoice', '=', False)]}"
```
You should hide it only if it is null, no need to check on the other. WDYT?

etobella replied:
Te same for the refund sequence :wink:

Shide replied:
Because when both fields (`self_invoice_sequence_id` and `self_invoice_refund_sequence_id`) are not set, only the button must be displayed. 
This is why is the same attrs on both fields.

etobella reviewed (approved):
Technical review.

Just a small comment.

:+1:

Shide commented:
> Could I recommend, please:
> 
> * [ ]  Make visible the button for sending the invoice in draft to the vendor
> 
> Why? Because for our general case, we send it in draft, the vendor confirm it by replying and then we validate.
> 
> A [requirement](https://www.sumup.com/es-es/facturas/glosario/facturacion-por-el-destinatario/) is
> 
> > Por cada factura emitida, el empresario o profesional ha de mostrar su aceptación, mediante un procedimiento que acordará previamente con su cliente.
> 
> Thanks!

Done

Recap:
Added to show button when invoice is not cancelled
Added "This Self-bill is not valid" when the invoice is in draft state on the footer 
Added ES missing translations

rafaelbn commented:
/ocabot merge minor

etobella commented:
@Shide @rafaelbn translation should be managed with weblate not directly here. Touching it outside from it can give us some errors :dizzy_face:
